instrumentation/google-http-client-1.19/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/googlehttpclient/GoogleHttpRequestInstrumentation.java

```diff
@@ -49,78 +50,108 @@ public void transform(TypeTransformer transformer) {
         this.getClass().getName() + "$ExecuteAsyncAdvice");
   }
 
+  public static class AdviceScope {
+
+    private static final VirtualField<HttpRequest, Context> virtualField =
+        VirtualField.find(HttpRequest.class, Context.class);
+    private final Context context;
+    private final Scope scope;
+
+    public AdviceScope(Context context, Scope scope) {
+      this.context = context;
+      this.scope = scope;
+    }
+
+    @Nullable
+    public static AdviceScope start(HttpRequest request) {
+
+      Context parentContext = currentContext();
+      if (!instrumenter().shouldStart(parentContext, request)) {
+        return null;
+      }
+
+      Context context = instrumenter().start(parentContext, request);
+      return new AdviceScope(context, context.makeCurrent());
+    }
+
+    public void end(HttpRequest request, HttpResponse response, Throwable throwable) {
```

Comment:
SImilar to previous PRs: Why not store `request` in `start` in the scope, to keep the `end()` signature simpler?